### ISSUE TITLE
Tesseract thresholding eliminates text on bright background colors

### ISSUE THREAD

**Issue by jbarlow83** [tesseract-ocr/tesseract#1990]
### Environment

* **Tesseract Version**: 4.0.0-rc1-8-g1bee homebrew build
* **Platform**: macOS High Sierra

### Current Behavior:

When tesseract is run on the attached image, the text on highlight backgrounds is missing from output.

![highlight](https://user-images.githubusercontent.com/1825843/46923056-40f46480-cfc7-11e8-921e-a0ecfdf023dc.png)

```
tesseract -c tessedit_write_images=true  _.png stdout
Not highlighted text
```

The thresholder blacks out the text (this is tessinput.tif):

![tessinput](https://user-images.githubusercontent.com/1825843/46923075-76994d80-cfc7-11e8-903e-68a27814edb3.png)

### Expected Behavior:

Thresholder should treat highlights as background so that Tesseract recognizes all of the text.

**Comment by amitdo** [comment]
I wonder why tesseract use its own otsu binarization and not leptonica's one.

You might want to test this image with leptonica's binarization options.

It also possible that your issue is cause by some filter done before or after the binarization.

**Comment by amitdo** [comment]
I used gimp for image thresholding and gave the binary image to tesseract.

Output:
```
Not highlighted text

Highlighted text

 

 

Highlighted background
```

**Comment by amitdo** [comment]
related: https://github.com/tesseract-ocr/tesseract/issues/242#issuecomment-191994220

**Comment by amitdo** [comment]
Here's Tesseract's otsu:

https://github.com/tesseract-ocr/tesseract/blob/57970443b42b/src/ccstruct/otsuthr.cpp

**Comment by zhangpy** [comment]
@amitdo Could you show me how to use gimp to change image thresholding? thanks

**Comment by amitdo** [comment]
Gimp is a GUI tool. You might want to try imagemagick which is command line tool instead.

**Comment by amitdo** [comment]
CC: @jbreiden

**Comment by Shreeshrii** [comment]
>I wonder why does tesseract use its own otsu binarization and not leptonica's one.

https://github.com/DanBloomberg/leptonica/blob/master/src/binarize.c

```
 *  ===================================================================
 *  Image binarization algorithms are found in:
 *    grayquant.c:   standard, simple, general grayscale quantization
 *    adaptmap.c:    local adaptive; mostly gray-to-gray in preparation
 *                   for binarization
 *    binarize.c:    special binarization methods, locally adaptive and
 *                   global.
 *  ================================================================
```

**Comment by amitdo** [comment]
With my patch in 3418, and the eng.traineddata from best I get:
```

Not highlighted text

Highlighted text

Highlighted background

```
I used `-c thresholding_method=2` -> Sauvola thresholding method.

**Comment by amitdo** [comment]
Fixed in #3418.
